### PR TITLE
chore(deps): Updated Kotlin from v1.7.0 to v1.9.10, Gradle from v6.7.…

### PR DESCRIPTION
**Pull request by drstrangelooker**
…1 to v8.3 and upgraded linter (#1343)

* chore(deps): Updated Kotlin from v1.7.0 to v1.9.10, Gradle from v6.7.1 to v8.3 and upgraded linter

* chore(deps): upgraded kotlin linter formatted code

* chore(deps): Updated Kotlin from v1.7.0 to v1.9.10, Gradle from v6.7.1 to v8.3 and upgraded linter

adding foojay-resolver-convention gradle plugin

* chore(deps): Updated Kotlin from v1.7.0 to v1.9.10, Gradle from v6.7.1 to v8.3 and upgraded linter

setting gradle jvm toolchain version from 8 to 17

* chore(deps): upgraded kotlin linter formatted code

from https://github.com/looker-open-source/sdk-codegen/pull/1343

### CODE REVIEW

**Comment by drstrangelooker** [comment]
@daniel-lopes-optimizely Can you fill out the cla agreement at https://cla.developers.google.com/? Thanks.

**Comment by daniel-lopes-optimizely** [comment]
@drstrangelooker I've already signed a CLA through my employer and the check was working previously for me.

Looking at the [failing check](https://github.com/looker-open-source/sdk-codegen/pull/1379/checks?check_run_id=17651535041) my author email isn't there anymore, it was replaced by the github reply email in the Merge commit.

**Comment by daniel-lopes-optimizely** [comment]
@drstrangelooker I've fixed the merge commit author, should work now.

**Comment by drstrangelooker** [comment]
@daniel-lopes-optimizely I went to my local branch and did `git commit --amend --no-edit --author "Daniel Lopes <snipped email>"`

Then I pushed the amended commit.

**Comment by drstrangelooker** [comment]
@tjbanghart ready for review

**Comment by daniel-lopes-optimizely** [comment]
hi @tjbanghart @drstrangelooker can we get this merged?

**Comment by drstrangelooker** [comment]
I'm reverting another change that hit main, then I'll get this merged.

**Comment by drstrangelooker** [comment]
@tjbanghart @daniel-lopes-optimizely Can you resolve these conflicts? Most are trivial but `kotlin/src/main/com/looker/rtl/Transport.kt` is non-trivial.

**Comment by daniel-lopes-optimizely** [comment]
@drstrangelooker @tjbanghart I've updated [my branch](https://github.com/daniel-lopes-optimizely/sdk-codegen/tree/kotlin-gradle-upgrade) with the latest changes from main

File: kotlin/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+dependencies {
+    implementation("org.jetbrains.kotlin:kotlin-reflect:$kotlinVersion")
+
+    implementation("io.github.cdimascio:dotenv-kotlin:6.2.2")
+    implementation("org.ini4j:ini4j:0.5.4")
+
+    implementation("commons-configuration:commons-configuration:1.10")
+
+    implementation(platform("com.google.http-client:google-http-client-bom:$googleHttpVersion"))
```
Comment (tjbanghart) [inline kotlin/build.gradle.kts:40]: Nice, thanks!

**Review by tjbanghart — APPROVED** [review]

**Pull request merged**